src/gui/qgsmapcanvas.cpp

```diff
@@ -585,13 +586,19 @@ void QgsMapCanvas::refreshMap()
     mSettings.setLayerStyleOverrides( QgsProject::instance()->mapThemeCollection()->mapThemeStyleOverrides( mTheme ) );
   }
 
+  // render main annotation layer above all other layers
+  QgsMapSettings renderSettings = mSettings;
+  QList<QgsMapLayer *> allLayers = renderSettings.layers();
+  allLayers.insert( 0, QgsProject::instance()->mainAnnotationLayer() );
```

Comment:
We'll need to apply this to the save map [canvas] as image/PDF.

Reply:
Good catch, will do